File: algorithms/sort/C/sorts.c
void bubblesort(int *, int);
void selectionsort(int *, int);
void insertionsort(int *, int);

int main() {
	return 0;
}

void bubblesort(int *arr, int len) {
	int i, j, aux;
	
	len /= 4;
	for (i = 0; i < len - 1; i++)
		for (j = i + 1; j < len; j++)
			if (*(arr + i) > *(arr + j)) {
				aux = *(arr + i);
				*(arr + i) = *(arr + j);
				*(arr + j) = aux;
			}
}

void selectionsort(int *arr, int len) {
	int i, j, min, aux;
	
	len /= 4;
	for (i = 0; i < len; i++) {
		min = i;
		for (j = i + 1; j < len; j++)
			if (*(arr + j) < *(arr + min)) min = j;
		aux = *(arr + i);
		*(arr + i) = *(arr + min);
		*(arr + min) = aux;
	}
}

void insertionsort(int *arr, int len) {
	int i, j, aux;
	
	len /= 4;
	for (i = 1; i < len; i++) {
		aux = *(arr + i);
		j = i - 1;
		while (j >= 0 && *(arr + j) > aux) {
			*(arr + (j + 1)) = *(arr + j);
			j--;
		}
		*(arr + (j + 1)) = aux;
	}
}

File: algorithms/sort/Python/sorts.py
class Bubble:
    
    def sort(seq):
        for i in range(len(seq) - 1):
            for j in range(i + 1, len(seq)):
                if seq[i] > seq[j]:
                    seq[i], seq[j] = seq[j], seq[i]

class Selection:
    
    def sort(seq):
        for i in range(len(seq)):
            min = i
            for j in range(i + 1, len(seq)):
                if seq[j] < seq[min]:
                    min = j
            seq[i], seq[min] = seq[min], seq[i]
            

class Insertion:
    
    def sort(seq):
        for i in range(1, len(seq)):
            aux = seq[i]
            j = i - 1
            while j >= 0 and seq[j] > aux:
                seq[j + 1] = seq[j]
                j -= 1
            seq[j + 1] = aux

class Shell:
    
    def sort(seq):
        h = 1
        while h < len(seq) // 3:
            h = (3 * h) + 1
        while h >= 1:
            for i in range(h, len(seq)):
                j = i
                while j >= h and seq[j - h] > seq[j]:
                    seq[j - h], seq[j] = seq[j], seq[j - h]
                    j -= h
            h = h // 3
            
class Merge:
    
    def sort(seq):
        if len(seq) < 30:
            Insertion.sort(seq)
            return seq
        mid = len(seq) // 2
        left, right = Merge.sort(seq[:mid]), Merge.sort(seq[mid:])
        return Merge.merge(left, right)
        
    def merge(left, right):
        result = []
        i = j = 0
        while i < len(left) and j < len(right):
            if left[i] < right[j]:
                result.append(left[i])
                i += 1
            else:
                result.append(right[j])
                j += 1
        result += left[i:]
        result += right[j:]
        return result